lib/Parser/Parser.cpp

```diff
@@ -164,7 +165,33 @@ Token Lexer::getNextToken(std::string &ErrStr) {
       ++Begin;
     } while (Begin != End && ((*Begin >= 'a' && *Begin <= 'z') ||
              (*Begin == '.')));
-    return Token{Token::Ident, TokenBegin, size_t(Begin - TokenBegin), APInt()};
+    // Parse the knownbits followed by "var"
+    const char *TokenEnd = Begin;
+    std::string IsKindNameVar = StringRef(TokenBegin, Begin - TokenBegin);
+    if (IsKindNameVar.compare("var") == 0) {
+      if (*Begin != '\n') {
+        ++Begin;
+        if (*Begin == '0' || *Begin == '1' || *Begin == 'x') {
+          const char *NumBegin = Begin;
+          while (Begin != End && (*Begin == '0' || *Begin == '1' || *Begin == 'x')) {
+            ++Begin;
+          }
+          const char *NumEnd = Begin - 1;
+          Token T;
+          T.K = Token::Ident;
+          T.Pos = TokenBegin;
+          T.Len = size_t(TokenEnd - TokenBegin);
+          T.PatternString = StringRef(NumBegin, NumEnd - NumBegin + 1);
+          return T;
+        } else {
+          return Token{Token::Ident, TokenBegin, size_t(TokenEnd - TokenBegin), APInt()};
+        }
+      } else {
+        return Token{Token::Ident, TokenBegin, size_t(TokenEnd - TokenBegin), APInt()};
+      }
+    } else {
+      return Token{Token::Ident, TokenBegin, size_t(TokenEnd - TokenBegin), APInt()};
+    }
```

Comment:
The reason why I had to do it here is because right after parsing the instruction names, we parse the token for integers '[-][0-9]:i[0-9]' or untyped integer '[0-9]' which come in 'block' instructions. So, for our case where I add the parsing for [01x], if the bit pattern is just 0 or 1, it would be considered untyped int token. That's why separate tokenization is not done.